exonum-java-binding-core/src/test/java/com/exonum/binding/storage/indices/ProofListContainsMatcher.java

```diff
@@ -67,19 +74,38 @@ public void describeTo(Description description) {
   protected void describeMismatchSafely(ProofListIndexProxy<String> list,
                                         Description mismatchDescription) {
     ListProof proof = proofFunction.apply(list);
-    ListProofRootHashCalculator<String> validator = newProofValidator();
+    ListProofRootHashCalculator<String> calculator = newProofHashCodeCalculator();
+    proof.accept(calculator);
```

Comment:
Why do we perform checks in the opposite order? I'd perform them in usual order and move calculator down to where it is needed.